controllers/etcd_controller.go

```diff
@@ -215,6 +215,8 @@ func (r *EtcdReconciler) reconcile(ctx context.Context, etcd *druidv1alpha1.Etcd
 		}
 	}
 
+	handleEtcdScaleDown := etcd.Spec.Replicas == 0 && etcd.Status.Replicas > 0 && etcd.Status.ReadyReplicas > 0
```

Comment:
Why is it dependent on status field? Status filed might not represent current state of system. It simply represent last reconciled state. Specifically in case of this field, there might be disparity between actual system state and recorded state under status.
So, simply check for `spec.replicase`.

Reply:
Scaledown logic should only be applied when etcd spec replicas is 0 but actual current replicas is 1. We can get this value only from status field, or else we'll have to do a manual check on the existence of the PVC and then decide to execute scaledown logic

Reply:
But then, what happens if the druid fails to update the status may be because of network issue or similar. And the spec.replica is scaled down to zero: It won't delete the PVC.
So, to prevent it, it's fine to do those check every time we find spec.replicas to zero.